near-sdk/src/types/vm_types.rs

```diff
@@ -51,4 +51,9 @@ impl From<PromiseResult> for VmPromiseResult {
 pub enum PromiseError {
     /// Promise result failed.
     Failed,
+    /// Promise succeeded but result length exceeded the limit.
+    TooLong(
```

Comment:
Didn't you want to also include parsing errors here, so `#[callback]` would never panic? Though, that would require to ensure that we are backward compatible in terms of the behavior, e.g. introducing a new `#[callback_new]` attribute for 5.x version and rename it in 6.x

Reply:
Also, maybe rename it to `promise_result_checked()` then?